packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -0,0 +1,103 @@
+import type { Request, RequestHandler } from 'express';
+import { cacheDecorator } from 'common-common/src/cacheDecorator';
+
+const oneBlock = 6; // typical Cosmos block time is ~6 seconds
+
+export const cosmosCacheRPC = (duration: number): RequestHandler => {
+  // RPC specific codes from cosmJS requests
+  const activeCodes = ['0801', '0802']; // ['DepositPerion', 'VotingPeriod']
+  const completedCodes = [
+    '0803', // 'Passed'
+    '0804', // 'Rejected'
+    '0805', // 'Failed'
+    '0803220a0a080000000000000087', // 'Passed - Osmosis'
+    '0803220a0a080000000000000100', // 'Passed - Osmosis'
+    '0803220a0a080000000000000189', // 'Passed - Osmosis'
+  ];
+
+  return async function cacheIfCacheworthyRPC(req, res, next) {
+    const body = parseReqBody(req);
+    const params = JSON.stringify(body?.params);
+    const method = body?.method;
+
+    if (/^(tx)/.test(method)) {
+      // TX Response: do not cache
+      return next();
+    } else if (/^(block|status)/.test(method)) {
+      // BLOCK CHECK: short cache
+      duration = oneBlock;
+    } else if (/^Query\/(Votes|TallyResult|Deposits)/.test(params)) {
+      // LIVE DATA: short cache
+      duration = oneBlock;
```

Comment:
I think a 5-minute cache would meet our goals better here. We care more about performance than vote tally accuracy. WDYT @jnaviask @NakulManchanda